SU2_CFD/include/output/filewriter/CSurfaceFVMDataSorter.hpp

```diff
@@ -67,13 +67,52 @@ class CSurfaceFVMDataSorter final: public CParallelDataSorter{
 
   /*!
    * \brief Get the global index of a point.
-   * \input iPoint - the point ID.
+   * \param[in] iPoint - the local renumbered point ID.
    * \return Global index of a specific point.
    */
   unsigned long GetGlobalIndex(unsigned long iPoint)  override{
+    if (iPoint > nParallel_Poin)
+      SU2_MPI::Error(string("Local renumbered iPoint ID ") + to_string(iPoint) +
+                     string(" is larger than max number of nodes ") + to_string(nParallel_Poin), CURRENT_FUNCTION);
+
     return Renumber2Global[iPoint];
   }
 
+  /*!
+   * \brief Get the beginning global renumbered node ID of the linear partition owned by a specific processor.
+   * \param[in] rank - the processor rank.
+   * \return The beginning global renumbered node ID.
+   */
+  unsigned long GetNodeBegin(unsigned short rank) override {
+    return nPoint_Recv[rank];
+  } 
+
+  /*!
+   * \brief Get the Processor ID a Point belongs to.
+   * \param[in] iPoint - global renumbered ID of the point
+   * \return The rank/processor number.
+   */
+  unsigned short FindProcessor(unsigned long iPoint) override {
+
+    for (unsigned short iRank = 1; iRank < size; iRank++){
+      if (nPoint_Recv[iRank] > iPoint){
+        return iRank - 1;
+      }
+    }
+    return size-1;
+  }
+
+  /*!
+   * \brief Get the ending global renumbered node ID of the linear partition owned by a specific processor.
+   * \param[in] rank - the processor rank.
+   * \return The ending global renumbered node ID.
+   */
+  unsigned long GetNodeEnd(unsigned short rank) override {
+    for (unsigned short iRank = 0; iRank < size-1; iRank++){
+      return nPoint_Recv[iRank+1]-1;
+    }
+    return nPoint_Recv[size];
+  }
```

Comment:
This is weird.

Reply:
Hmm indeed. That looks like a strange loop. Cannot remember why we did it like that. Gonna check that.

Reply:
I am also a bit clueless what happened there, but that function never got called anyway... which kindof makes it even worse. So I removed it and also two additional useless code-lines.

Reply:
And I made the all "Get" functions I added `const` where possible